sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -130,19 +133,28 @@ package object util {
     ret
   }
 
+  // Replaces attributes, string literals, complex type extractors with their pretty form so that
+  // generated column names don't contain back-ticks or double-quotes.
+  def usePrettyExpression(e: Expression): Expression = e transform {
+    case a: Attribute => new PrettyAttribute(a)
+    case Literal(s: UTF8String, StringType) => PrettyAttribute(s.toString, StringType)
+    case e @ GetStructField(child, _, Some(name)) => PrettyGetStructField(child, name, e.dataType)
```

Comment:
can we also turn `GetStructField` to `PrettyAttribute` here? We can generate the sql string here and pass it as the name of `PrettyAttribute`